### PR TITLE
feat(server): server-side events

### PR DESCRIPTION
**Pull request by jrasm91**
Reliably send events to connected immich-server and immich-microservice containers when the system-config changes. Replaces old solution of using a single BullMQ job.

### CODE REVIEW

File: server/src/domain/system-config/system-config.service.spec.ts [hunk 1]
```diff
@@ -269,13 +262,14 @@ describe(SystemConfigService.name, () => {
...
+    it('should update the config and emit client and server events', async () => {
       configMock.load.mockResolvedValue(updates);
 
       await expect(sut.updateConfig(updatedConfig)).resolves.toEqual(updatedConfig);
 
+      expect(communicationMock.broadcast).toHaveBeenCalled();
+      expect(communicationMock.sendServerEvent).toHaveBeenCalledWith(ServerEvent.CONFIG_UPDATE);
       expect(configMock.saveAll).toHaveBeenCalledWith(updates);
-      expect(jobMock.queue).toHaveBeenCalledWith({ name: JobName.SYSTEM_CONFIG_CHANGE });
     });
```
Comment (bo0tzz) [inline server/src/domain/system-config/system-config.service.spec.ts:273]: Can this test just connect to the channel and assert it receives the message?

**Review by bo0tzz — APPROVED** [review]

File: server/src/domain/system-config/system-config.service.spec.ts (hunk 1 above)
Comment (zackpollard) [reply server/src/domain/system-config/system-config.service.spec.ts:273]: If this is a unit test I think this is the right level to mock and assert. We should have more tests on the event emitters themselves to ensure that we are confident that code works.

**Review by danieldietzler — APPROVED** [review]
That's nice!

File: server/src/domain/system-config/system-config.service.spec.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/domain/system-config/system-config.service.spec.ts:273]: It is a unit test and I think it makes sense like we have it. e2e will definitely take some more effort to check in an automated fashion, but I have at least manually verified that this is working correctly.

**Pull request merged**